src/nnue/nnue_feature_transformer.h

```diff
@@ -146,6 +147,70 @@ using psqt_vec_t = int32x4_t;
 #endif
 
 
+class Packing {
+   private:
+    // Store the order by which 128-bit blocks of a 1024-bit data must
+    // be permuted so that calling packus on adjacent vectors of 16-bit
+    // integers loaded from the data results in the pre-permutation order
+    static constexpr auto packus_epi16_order = []() -> std::array<std::size_t, 8> {
+#if defined(USE_AVX512)
+        // _mm512_packus_epi16 after permutation:
+        //  0     2     4     6     // Vector 0
+        //     1     3     5     7  // Vector 1
+        //  0  1  2  3  4  5  6  7  // Packed Result
```

Comment:
I'd suggest formatting it as 

```
// |   0   |   2   |   4   |   6   | // Vector 0
// |   1   |   3   |   5   |   7   | // Vector 1
// | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | // Packed Result
```

to make the value widths and placements clear